constantine/math/constants/banderwagon.nim

```diff
@@ -0,0 +1,69 @@
+# Constantine
+# Copyright (c) 2018-2019    Status Research & Development GmbH
+# Copyright (c) 2020-Present Mamy André-Ratsimbazafy
+# Licensed and distributed under either of
+#   * MIT license (license terms in the root directory or at http://opensource.org/licenses/MIT).
+#   * Apache v2 license (license terms in the root directory or at http://www.apache.org/licenses/LICENSE-2.0).
+# at your option. This file may not be copied, modified, or distributed except according to those terms.
+
+import
+  # Internals
+  ../../platforms/abstractions,
+  ../config/curves,
+  ../arithmetic,
+  ../extension_fields,
+  ../elliptic/ec_twistededwards_projective
+
+func `==`*(P, Q: ECP_TwEdwards_Prj[Fp[Banderwagon]]): SecretBool =
+  ## Equality check for points in the Banderwagon Group
+  ## The equals method is different for the quotient group
+  ## 
+  ## Check for the (0,0) point, which is possible
+  ## 
+  ## This is a costly operation
+
+  var lhs{.noInit.}, rhs{.noInit.}: typeof(P).F
+
+  # Check for the zero points
+  result = not(P.x.is_zero() and P.y.is_zero())
+  result = result or not(Q.x.is_zero() and Q.y.is_zero())
+
+  ## Check for the equality of the points
+  ## X1 * Y2 == X2 * Y1
+  lhs.prod(P.x, Q.y)
+  rhs.prod(Q.x, P.y)
+  result = result and lhs == rhs
```

Comment:
Interesting, it sems like this formula avoiding using Z coordinates and would be faster than the original one:

https://github.com/mratsim/constantine/blob/d51699248db04e29c7b1ad97e0bafa1499db00b5/constantine/math/elliptic/ec_twistededwards_projective.nim#L35-L47

Might be intersting to change the original and check the tests passes

Reply:
I think we can try this out separately. And keep that out of this PR

Reply:
The original formulas should still work for Banderwagon unless I missed something though so this shouldn't be needed.

On a side note if someone imports elliptic/ec_twistededwards_projective.nim and does not import constants/banderwagon.nim (or it's not reexported) the wrong `==` will be called.

So the file should import/reexport it to ensure the specialization is called.